lib/src/component/error_boundary.dart

```diff
@@ -309,6 +348,7 @@ class ErrorBoundaryComponent<T extends ErrorBoundaryProps, S extends ErrorBounda
     if (props.fallbackUIRenderer != null) {
       _lastError = error; // [1.1]
       _lastErrorInfo = _getReadableErrorInfo(jsErrorInfo); // [1.1]
+      _logErrorCaughtByErrorBoundary(error, _getReadableErrorInfo(jsErrorInfo)); // [3]
```

Comment:
We had originally discussed doing this logging as part of a default `onComponentDidCatch` implementation.

I'm not opposed to having it more built-in like it is now, but I think we should have a prop flag to be able to turn off logging.

Reply:
It might be nice to be able to use a non-hierarchical logger (e.g., `Logger.detached`), too... Perhaps 3 props?

```dart
bool shouldLog;
String loggerName;
Logger logger;
```

```dart
Logger _logger([TProps props]) {
  props ??= this.props;
  return props.shouldLog
      ? props.logger ?? Logger(props.loggerName) 
      : null;
}
```